specification/hybridnetwork/resource-manager/Microsoft.HybridNetwork/preview/2022-01-01-preview/vendor.json

```diff
@@ -0,0 +1,1219 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Creation or update is successful. The operation returns the resulting vendor sku sub resource.",
+            "schema": {
+              "$ref": "#/definitions/VendorSku"
+            }
+          },
+          "201": {
+            "description": "Created. The operation returns the resulting vendor sku sub resource.",
+            "schema": {
+              "$ref": "#/definitions/VendorSku"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create or update the sku of vendor resource": {
+            "$ref": "./examples/VendorSkuCreate.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.HybridNetwork/vendors/{vendorName}/vendorSkus": {
+      "get": {
+        "tags": [
+          "VendorSkus"
+        ],
+        "operationId": "VendorSkus_List",
+        "description": "Lists all the skus of a vendor.",
+        "parameters": [
+          {
+            "name": "vendorName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the vendor."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request is successful. The operation returns a list of vendor sku resources.",
+            "schema": {
+              "$ref": "#/definitions/VendorSkuListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List all the vendor skus of vendor resource": {
+            "$ref": "./examples/VendorSkuListByVendor.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.HybridNetwork/vendors/{vendorName}/vendorSkus/{skuName}/generateCredential": {
```

Comment:
>generateCredential [](http://example.com/codeflow?start=113&length=18)

Note if you want to use this in an ARM template then this needs to be called as list*

You can see other examples here : [https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#list](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#list)

Reply:
Thank you for the suggestion. We do not need to use it in an ARM template. Is it a strict requirement to call it as "list*"? If yes, I think we need to update the APIs in the private repo as well.

Reply:
Well once these are released then customers might want to deploy them in ARM templates since all azure resources are typically deployed using ARM templates.  For example if they want to generate the credential and then pass it into another deploymentscript to configure things. It is a strict requirement to call it list* only is you need to call it in ARM template otherwise it is fine.

Reply:
Thank you for the explaination. Renamed it to "listCredential".